packages/webgl/src/shader/shader.js

```diff
@@ -0,0 +1,249 @@
+import { Utils, Logger } from "scarlett-commons";
+import { GLU } from "../webglUtils";
+
+/**
+ * Shader Class
+ * Some cool code ideas were applied from Pixi.JS Shader class
+ *
+ * Relevant information regarding attribute usage:
+ * https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/vertexAttribPointer
+ */
+export default class Shader {
+  //#region Constructors
+
+  /**
+   *
+   * @param {WebGLRenderContext} glc Webgl Render Context
+   * @param {String} vertexScript
+   * @param {String} fragmentScript
+   * @param {*} uniforms
+   * @param {*} attributes
+   */
+  constructor(renderContext, vertexScript, fragmentScript, uniforms, attributes) {
+    if (!Utils.isObjectAssigned(vertexScript) || !Utils.isObjectAssigned(fragmentScript)) {
```

Comment:
We should check if renderContext is a WebGLRenderContext (or at least a RenderContext) as we don't have a fallback.